### PR TITLE
enable TLSRoute service reference in serviceTriggersRebuild

### PR DESCRIPTION
**Pull request by vmw-yingy**
enable DAG rebuild for service referenced by TLSRoutes

Fixes: #4831

### CODE REVIEW

**Comment by vmw-yingy** [comment]
@skriss @sunjayBhatia Please help review

**Review by sunjayBhatia — APPROVED** [review]

**Comment by sunjayBhatia** [comment]
One last thing before we merge this @vmw-yingy, it would be create if you could add a changelog file following the guidelines here: https://github.com/projectcontour/contour/blob/main/CONTRIBUTING.md#commit-message-and-pr-guidelines, in this case a `small` changelog

File: changelogs/unreleased/4915-yingy-small.md
Comment (sunjayBhatia) [inline changelogs/unreleased/4915-yingy-small.md:1]: suggestion on the wording, to make it more user focused
```suggestion
Ensure changes to Services referenced by TLSRoutes trigger xDS configuration updates.
```

File: changelogs/unreleased/4915-yingy-small.md
Comment (sunjayBhatia) [inline changelogs/unreleased/4915-yingy-small.md:1]: oh oops, didnt catch it but the filename should be `4915-vmw-yingy-small.md` to get your full github username in

**Review by sunjayBhatia — CHANGES_REQUESTED** [review]
changelog filename needs an update

**Review by skriss — COMMENTED** [review]
LGTM once the changelog filename is updated.

I'm thinking we can hold off on merging until we merge the `dev-gwapi-0.6` branch into `main` (which should be soon, the GWAPI 0.6.0 release should be out this week or next). Would keep the commit history a little cleaner.

**Review by sunjayBhatia — APPROVED** [review]

**Review by skriss — APPROVED** [review]

**Comment by skriss** [comment]
Now that we've merged the `dev-gwapi-0.6` branch into `main`, we'll want to rebase this PR onto `main` and target `main` to get it merged. We can walk through this together on Friday.

**Comment by skriss** [comment]
I re-targeted the PR to main; the branch now needs to be rebased on upstream/main.

**Pull request merged**
